### PR TITLE
Wayland: Add generic deleted_flag_for_resource() helper

### PR DESCRIPTION
**Pull request by RAOF**

### CODE REVIEW

**Comment by RAOF** [comment]
Ping @wmww - here's the generic solution.

**Comment by wmww** [comment]
bors cancel

**Comment by wmww** [comment]
Is there a reason this is nested in the wayland namespace? generally that's just for generated code and our stuff goes in the frontend namespace.

File: src/server/frontend_wayland/deleted_for_resource.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <memory>
+
+struct wl_resource;
+
+namespace mir
+{
+namespace frontend
+{
+namespace wayland
```
Comment (wmww) [inline src/server/frontend_wayland/deleted_for_resource.h]: Is there a reason this is nested in the wayland namespace? generally that's just for generated code and our stuff goes in the frontend namespace.

**Pull request merged**